### PR TITLE
Refactor makefile to support multiple toolchains

### PR DESCRIPTION
**Pull request by theotherjimmy**
To test it I added a make-armcc toolchain

### CODE REVIEW

**Comment by 0xc0170** [comment]
Interesting, armcc makefile :-)  I'll have a look this weekend at this

**Comment by theotherjimmy** [comment]
The best part is that it works.

**Comment by 0xc0170** [comment]
One request - having a test for new tool class?

File: project_generator/tools/tool.py [hunk 1]
```diff
@@ -15,8 +15,9 @@
 import os
 import logging
 
-from os.path import join, dirname
-from jinja2 import Template
+from os.path import join, dirname, abspath
```
Comment (0xc0170) [inline project_generator/tools/tool.py]: nitpick: this change should be a separate commit with a description of using jinja enviroment and file loader

**Comment by 0xc0170** [comment]
Looks good ! Please add a test for new armcc makefile class, and we shall be good to go !

**Comment by 0xc0170** [comment]
Any update @theotherjimmy ?

File: project_generator/tools/tool.py (hunk 1 above)
Comment (theotherjimmy) [inline project_generator/tools/tool.py]: I'll rewrite history to make this happen.

**Comment by theotherjimmy** [comment]
Working on the test soon.

File: project_generator/tools/tool.py (hunk 1 above)
Comment (theotherjimmy) [inline project_generator/tools/tool.py]: There you go, now no one has the context to understand why you made this comment. :P
